pkg/csi/service/common/common_controller_helper.go

```diff
@@ -141,6 +143,69 @@ func CheckAPI(version string) error {
 	return nil
 }
 
+// UseVslmAPIs checks if specified version is between 6.7 Update 3l and 7.0
+// The method takes aboutInfo{} as input which contains details about
+// VC version, build number and so on.
+// If the version is between the upper and lower bounds, the method returns true, else returns false
+// and appropriate errors during failure cases
+func UseVslmAPIs(ctx context.Context, aboutInfo vim25types.AboutInfo) (bool, error) {
+	log := logger.GetLogger(ctx)
+	items := strings.Split(aboutInfo.ApiVersion, ".")
+	apiVersion := strings.Join(items[:], "")
+	// Convert version string to string, Ex: "6.7.3" becomes 673, "7.0.0.0" becomes 700
```

Comment:
"7.0.0.0" becomes 700? so are we skipping the last version number?

Reply:
We can safely ignore 4th digit. It is not significant, see below for difference between 7.0 and 7.0U1

```
VC 7.0
cnsClient Version: (types.AboutInfo) {
         DynamicData: (types.DynamicData) {
         },
         Name: (string) (len=21) "VMware vCenter Server",
         FullName: (string) (len=42) "VMware vCenter Server 7.0.0 build-15952599",
         Vendor: (string) (len=12) "VMware, Inc.",
         Version: (string) (len=5) "7.0.0",
         Build: (string) (len=8) "15952599",
         LocaleVersion: (string) (len=4) "INTL",
         LocaleBuild: (string) (len=3) "000",
         OsType: (string) (len=9) "linux-x64",
         ProductLineId: (string) (len=3) "vpx",
         ApiType: (string) (len=13) "VirtualCenter",
         ApiVersion: (string) (len=7) "7.0.0.0",
         InstanceUuid: (string) (len=36) "b71e9731-7e4f-423d-9659-7f957cb44355",
         LicenseProductName: (string) (len=27) "VMware VirtualCenter Server",
         LicenseProductVersion: (string) (len=3) "7.0"
        }



VC 70U1

cnsClient Version: (types.AboutInfo) {
         DynamicData: (types.DynamicData) {
         },
         Name: (string) (len=21) "VMware vCenter Server",
         FullName: (string) (len=42) "VMware vCenter Server 7.0.1 build-16858589",
         Vendor: (string) (len=12) "VMware, Inc.",
         Version: (string) (len=5) "7.0.1",
         Build: (string) (len=8) "16858589",
         LocaleVersion: (string) (len=4) "INTL",
         LocaleBuild: (string) (len=3) "000",
         OsType: (string) (len=9) "linux-x64",
         ProductLineId: (string) (len=3) "vpx",
         ApiType: (string) (len=13) "VirtualCenter",
         ApiVersion: (string) (len=7) "7.0.1.1",
         InstanceUuid: (string) (len=36) "87a49371-bd05-4088-b895-6ffe408beef6",
         LicenseProductName: (string) (len=27) "VMware VirtualCenter Server",
         LicenseProductVersion: (string) (len=3) "7.0"
        }
```